File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/ModelViewTest.java
package com.raizlabs.android.dbflow.test.structure;

import com.raizlabs.android.dbflow.list.FlowCursorList;
import com.raizlabs.android.dbflow.sql.language.Select;
import com.raizlabs.android.dbflow.test.FlowTestCase;

import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

/**
 * Description:
 */
public class ModelViewTest extends FlowTestCase {

    /**
     * Tests to ensure the model view operates as expected
     */
    @Test
    public void testModelView() {
        TestModel2 testModel2 = new TestModel2();
        testModel2.order = 6;
        testModel2.name = "View";
        testModel2.save();

        testModel2 = new TestModel2();
        testModel2.order = 5;
        testModel2.name = "View2";
        testModel2.save();

        List<TestModelView> testModelViews = new Select().from(TestModelView.class).queryList();
        assertTrue(!testModelViews.isEmpty());
        assertTrue(testModelViews.size() == 1);
    }

    @Test
    public void testModelViewCursorList() {
        FlowCursorList<TestModelView> list = new FlowCursorList<>(true, TestModelView.class);
        assertNotNull(list.getItem(0));
    }

}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/ModelViewTest.kt
package com.raizlabs.android.dbflow.test.structure

import com.raizlabs.android.dbflow.list.FlowCursorList
import com.raizlabs.android.dbflow.sql.language.SQLite
import com.raizlabs.android.dbflow.sql.language.Select
import com.raizlabs.android.dbflow.test.FlowTestCase

import org.junit.Test

import org.junit.Assert.assertNotNull
import org.junit.Assert.assertTrue

/**
 * Description:
 */
class ModelViewTest : FlowTestCase() {

    /**
     * Tests to ensure the model view operates as expected
     */
    @Test
    fun testModelView() {
        var testModel2 = TestModel2()
        testModel2.order = 6
        testModel2.name = "View"
        testModel2.save()

        testModel2 = TestModel2()
        testModel2.order = 5
        testModel2.name = "View2"
        testModel2.save()

        val testModelViews = Select().from(TestModelView::class.java).queryList()
        assertTrue(!testModelViews.isEmpty())
        assertTrue(testModelViews.size == 1)

        val list = FlowCursorList.Builder(
                SQLite.select().from(TestModelView::class.java))
                .build()
        assertNotNull(list.getItem(0))
    }

}
